### ISSUE TITLE
Jax is 50 times slower on my machine

### ISSUE THREAD

**Issue by smy20011** [google/jax#2143]
Hey, I just started using Jax for my personal project and notice a huge drop of performance. However, when I run same code on Colab, Jax's speed is comparable to numpy.

Do you know any way to debug this performance issue? Thank you!

Local Jax Perforamnce:

![Jax local](https://user-images.githubusercontent.com/1560175/73618643-df05a700-45dd-11ea-837d-ed08580d0d92.png)

Jax performance on colab:

![Jax colab](https://user-images.githubusercontent.com/1560175/73618642-df05a700-45dd-11ea-8d02-ec66055e7d95.png)

**Comment by smy20011** [comment]
It seems happens after 0.1.56

**Comment by smy20011** [comment]
Colab to reproduce it.
https://colab.research.google.com/drive/1i9y8K6gywXGDarpNuVkDMQeLrVYgE5nG

**Comment by shoyer** [comment]
Colab is probably still running an older version of JAX, which would explain the difference.

Are you seeing a slow-down in real code examples? Creating a small array and performing multiplication is doing a very small amount of computation, so you are mostly seeing JAX's overhead here. Also note that you need to add `.block_until_ready()` to measure JAX's real performance here -- see https://jax.readthedocs.io/en/latest/async_dispatch.html for more details.

**Comment by hawkinsp** [comment]
Another thing I'll point out is that `arange` is now lazy in current versions of JAX. So you aren't actually constructing the array at all, merely building some Python data structures.

Microbenchmarks are useful tools, but they can be misleading. Do you have a more realistic benchmark?

**Comment by smy20011** [comment]
Hey, sorry for the late reply. I just updated my colab with my code to demonstrate the performance issue.

https://colab.research.google.com/drive/1i9y8K6gywXGDarpNuVkDMQeLrVYgE5nG#scrollTo=FF0zXWVqoh-3

**Comment by shoyer** [comment]
To get good performance in JAX, you generally need to use `jit`. I don't see that in your code.

**Comment by smy20011** [comment]
Thank you shoyer! It indeed faster on the benchmark with jit. However, when I take the gradient of the function and @jit it. It seems Jax cannot handle naive python loop in the code. 

![Screenshot 2020-02-08 at 09 50 54](https://user-images.githubusercontent.com/1560175/74089718-e3581700-4a58-11ea-8eee-0c6ec8dda358.png)

Should I use lax.fori_loop instead? Thank you!

**Comment by shoyer** [comment]
There are a handful of options here, I recommend reading through https://jax.readthedocs.io/en/latest/notebooks/Common_Gotchas_in_JAX.html#%F0%9F%94%AA-Control-Flow

**Comment by smy20011** [comment]
Great! Thank you! Closing this bug as resolved.
